opennlp-dl/src/test/java/opennlp/dl/doccat/DocumentCategorizerDLEval.java

```diff
@@ -92,6 +92,46 @@ public void categorize() throws IOException, OrtException {
 
   }
 
+  @Test
+  public void categorizeWithAutomaticLabels() throws IOException, OrtException {
+
+    final File model = new File(getOpennlpDataDir(),
```

Comment:
Does OpenNLP package these, or do you download them to run the tests?  A question taht will come up in Solr-land is "Wait, for the test you want a 600 mb .onnx file???"...  ;-).  How do you handle the models for testing?

Reply:
No. In this configuration, it relies on a data directory, which must supplied via a `-DOPENNLP_DATA_DIR=PATH` parameter or system property value.

There are other options used in OpenNLP as well:

- https://github.com/apache/opennlp/blob/main/opennlp-tools/src/main/java/opennlp/tools/util/DownloadUtil.java#L126 is used to download models. Models are cached in a local directory on disk
- One could also rely on maven download plugin to download and cache locally.
- https://github.com/apache/opennlp/blob/6fde608cb0dd5866c6330f3e3dcd04f791c4ef96/opennlp-tools/src/test/java/opennlp/tools/EnabledWhenCDNAvailable.java#L38 only download stuff, if you are online

I could also imagine, that one could build a special annotation to only download the model files, if you are in a CI/CD context in which it doesn't really matter, if additional 600mb are downloaded ;-)

Reply:
For our evaluation tests in OpenNLP, we are downloading our whole evaluation dataset (~3 GB) from nightlies.apache.org for each run and use that content for the `OPENNLP_DATA_DIR`. So it doesn't really matter ;-) - I could imagine, that it might be difficult to argue in a project like SOLR.